core/src/main/java/org/apache/iceberg/stats/StatsReader.java

```diff
@@ -0,0 +1,183 @@
+            "Invalid footer size: %s", size));
+  }
+
+  public FileMetadata getFileMetadata() throws IOException {
+    if (!knownFileMetadata.isPresent()) {
+      int footerSize = getFooterSize();
+      byte[] footer = readInput(fileLength - footerSize, footerSize);
+
+      checkMagic(footer, 0);
+      checkMagic(footer, footerSize - 4);
+
+      int fileFormatVersion = StatsFormat.readIntegerLittleEndian(footer, footerSize - 8);
+      Preconditions.checkState(
+          fileFormatVersion == StatsFormat.CURRENT_FORMAT_VERSION,
+          "Unsupported format version %s, %s is the latest supported",
+          fileFormatVersion,
+          StatsFormat.CURRENT_FORMAT_VERSION);
+
+      int flags = StatsFormat.readIntegerLittleEndian(footer, footerSize - 12);
+      Preconditions.checkState((flags & ~StatsFormat.SUPPORTED_FLAGS) == 0, "Unsupported flags: %s", flags);
+      boolean compressed = isFlagSet(flags, StatsFormat.FLAG_COMPRESSED);
+
+      int reserved = StatsFormat.readIntegerLittleEndian(footer, footerSize - 16);
+      Preconditions.checkState(reserved == 0, "Unexpected reserved bytes value: %s", reserved);
+
+      int footerPayloadSize = StatsFormat.readIntegerLittleEndian(footer, footerSize - 20);
+      Preconditions.checkState(footerPayloadSize == footerSize - 24,
+          "Unexpected footer payload size value %s for footer size %s", footerPayloadSize, footerSize);
+
+      FileMetadata readFileMetadata;
+      ByteBuffer footerPayload = ByteBuffer.wrap(footer, 4, footerPayloadSize);
+      if (!compressed) {
+        readFileMetadata = parseFileMetadata(footerPayload);
+      } else {
+        ByteBuffer footerJson = StatsFormat.decompressFooterPayload(footerPayload);
+        readFileMetadata = parseFileMetadata(footerJson);
+      }
+      this.knownFileMetadata = Optional.of(readFileMetadata);
+    }
+    return knownFileMetadata.get();
+  }
+
+  /**
+   * @throws IOException          when I/O error occurs
+   * @throws UncheckedIOException when I/O error occurs
+   */
+  // Note: The method is marked as throwing IOException to allow future implementation evolution.
+  // Currently, the exception isn't being thrown.
+  public Stream<Map.Entry<BlobMetadata, ByteBuffer>> readAll(List<BlobMetadata> blobs) throws IOException {
+    if (blobs.isEmpty()) {
+      return Stream.empty();
+    }
+
+    // TODO inspect blob offsets and coalesce read regions close to each other
+
+    return blobs.stream()
+        .sorted(Comparator.comparingLong(BlobMetadata::offset))
+        .map((BlobMetadata blobMetadata) -> {
+          try {
+            input.seek(blobMetadata.offset());
+            byte[] bytes = new byte[Math.toIntExact(blobMetadata.length())];
+            ByteStreams.readFully(input, bytes);
+            ByteBuffer data = ByteBuffer.wrap(bytes);
+            if (blobMetadata.compressionCodec() != null) {
+              StatsCompressionCodec codec = StatsCompressionCodec.forName(blobMetadata.compressionCodec());
```

Comment:
Why not translate to/from `StatsCompressionCodec` in the parser? That would make more sense to me.

Reply:
Forward compatibility for compression codecs. If we add e.g. LZ6, old reader can still make use of the file, it just cannot read portions of the file.

Reply:
Yes, I can add `StatsCompressionCodec.UNKOWN`. To me it feels like a special value, very much like `-1`, which I would rather avoid. What would be the benefit from such a change? 

If the user code was interested in compression information, a String is better than an UNKNOWN constant, since String conveys all information, while UNKNOWN already lost something.
If the user code is not dealing with compression information directly, it doesn't matter, it's an impl detail.